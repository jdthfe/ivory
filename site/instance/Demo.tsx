import React from 'react';
import { getInstance } from '../_util';

interface DemoProps {
    item: ComponentIndex;
}
const Demo = (props: DemoProps) => {
    const { item } = props;
    const Demo = getInstance(item) || null;
    return <main className="instance-demo">{Demo ? <Demo /> : null}</main>;
};

export default Demo;
