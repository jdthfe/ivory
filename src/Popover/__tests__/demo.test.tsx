import webDemoTest from '@tests/shared/demoTest';
webDemoTest('Popover');

import React from 'react';
import Demo from '../demo';
import { render, fireEvent, waitForElement, waitForElementToBeRemoved } from 'react-testing-library';

describe('PopoverDemo', () => {
    it('Show Popover bottom-center', async () => {
        const { baseElement, getAllByRole } = render(<Demo />);
        const btn = getAllByRole('button')[0];
        fireEvent.click(btn);
        await waitForElement(() => baseElement.querySelector(`.eui-popover.eui-fade-entry-done.eui-fade`));
        await waitForElement(() =>
            baseElement.querySelector(`.eui-cover.eui-cover-transparent.eui-fade-entry-done.eui-fade`),
        );

        expect(baseElement).toMatchSnapshot('visible = true');

        const cover = baseElement.querySelector('.eui-cover');
        if (cover) {
            fireEvent.click(cover);
        }
        await waitForElement(() => baseElement.querySelector(`.eui-fade-exit-active`));
        await waitForElementToBeRemoved(() => baseElement.querySelector(`.eui-fade-exit-active`));

        expect(baseElement).toMatchSnapshot('visible = false');
    });

    it('Show Popover top-center', async () => {
        const { baseElement, getAllByRole } = render(<Demo />);
        const btn = getAllByRole('button')[1];
        fireEvent.click(btn);
        await waitForElement(() => baseElement.querySelector(`.eui-popover.eui-fade-entry-done.eui-fade`));
        await waitForElement(() =>
            baseElement.querySelector(`.eui-cover.eui-cover-transparent.eui-fade-entry-done.eui-fade`),
        );

        expect(baseElement).toMatchSnapshot('visible = true');

        const cover = baseElement.querySelector('.eui-cover');
        if (cover) {
            fireEvent.click(cover);
        }
        await waitForElement(() => baseElement.querySelector(`.eui-fade-exit-active`));
        await waitForElementToBeRemoved(() => baseElement.querySelector(`.eui-fade-exit-active`));

        expect(baseElement).toMatchSnapshot('visible = false');
    });
});
