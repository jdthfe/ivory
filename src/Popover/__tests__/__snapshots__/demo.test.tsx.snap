// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverDemo Show Popover bottom-center: visible = false 1`] = `
<body>
  <div>
    <div
      class="eui-wingblank eui-wingblank-m"
    >
      <div>
        <button
          class="eui-button eui-button-l eui-button-customize eui-button-ghost"
          role="button"
        >
          Popover
        </button>
      </div>
      <div
        class="eui-whitespace eui-whitespace-xl"
      />
      <div>
        <button
          class="eui-button eui-button-l eui-button-customize eui-button-ghost"
          role="button"
        >
          Popover
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`PopoverDemo Show Popover bottom-center: visible = true 1`] = `
<body>
  <div>
    <div
      class="eui-wingblank eui-wingblank-m"
    >
      <div>
        <button
          class="eui-button eui-button-l eui-button-customize eui-button-ghost"
          role="button"
        >
          Popover
        </button>
      </div>
      <div
        class="eui-whitespace eui-whitespace-xl"
      />
      <div>
        <button
          class="eui-button eui-button-l eui-button-customize eui-button-ghost"
          role="button"
        >
          Popover
        </button>
      </div>
    </div>
  </div>
  <div
    class="eui-cover eui-cover-transparent eui-fade-entry-done eui-fade"
  />
  <div
    class="eui-popover eui-fade-entry-done eui-fade"
  >
    <div
      class="eui-popover-wrap"
      style="top: 4px; left: 0px; transform-origin: 0px 0 0;"
    >
      <div
        class="eui-popover-content"
      >
        <div
          class="eui-popover-arrow eui-popover-arrow-bottom"
        />
        <div
          class="eui-popover-inner"
        >
          <div
            class="eui-popover-inner-wrapper"
          >
            <p
              style="margin: 10px 20px;"
            >
              red
            </p>
            <p
              style="margin: 10px 20px;"
            >
              yellow
            </p>
            <p
              style="margin: 10px 20px;"
            >
              blue
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`PopoverDemo Show Popover top-center: visible = false 1`] = `
<body>
  <div>
    <div
      class="eui-wingblank eui-wingblank-m"
    >
      <div>
        <button
          class="eui-button eui-button-l eui-button-customize eui-button-ghost"
          role="button"
        >
          Popover
        </button>
      </div>
      <div
        class="eui-whitespace eui-whitespace-xl"
      />
      <div>
        <button
          class="eui-button eui-button-l eui-button-customize eui-button-ghost"
          role="button"
        >
          Popover
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`PopoverDemo Show Popover top-center: visible = true 1`] = `
<body>
  <div>
    <div
      class="eui-wingblank eui-wingblank-m"
    >
      <div>
        <button
          class="eui-button eui-button-l eui-button-customize eui-button-ghost"
          role="button"
        >
          Popover
        </button>
      </div>
      <div
        class="eui-whitespace eui-whitespace-xl"
      />
      <div>
        <button
          class="eui-button eui-button-l eui-button-customize eui-button-ghost"
          role="button"
        >
          Popover
        </button>
      </div>
    </div>
  </div>
  <div
    class="eui-cover eui-cover-transparent eui-fade-entry-done eui-fade"
  />
  <div
    class="eui-popover eui-fade-entry-done eui-fade"
  >
    <div
      class="eui-popover-wrap"
      style="top: -4px; left: 0px; transform-origin: 0px 0 0;"
    >
      <div
        class="eui-popover-content"
      >
        
        <div
          class="eui-popover-inner"
        >
          <div
            class="eui-popover-inner-wrapper"
          >
            <p
              style="margin: 10px 20px;"
            >
              red
            </p>
            <p
              style="margin: 10px 20px;"
            >
              yellow
            </p>
            <p
              style="margin: 10px 20px;"
            >
              blue
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders Popover correctly 1`] = `
<div>
  <div
    class="eui-wingblank eui-wingblank-m"
  >
    <div>
      <button
        class="eui-button eui-button-l eui-button-customize eui-button-ghost"
        role="button"
      >
        Popover
      </button>
    </div>
    <div
      class="eui-whitespace eui-whitespace-xl"
    />
    <div>
      <button
        class="eui-button eui-button-l eui-button-customize eui-button-ghost"
        role="button"
      >
        Popover
      </button>
    </div>
  </div>
</div>
`;
